pallets/xcm-transactor/src/lib.rs

```diff
@@ -862,71 +854,30 @@ pub mod pallet {
 			Ok(dest)
 		}
 
-		/// Returns weight of `transact_through_derivative` call.
-		fn weight_of_transact_through_derivative(
-			index: &u16,
-			dest: &T::Transactor,
-			weight: &u64,
-			call: &[u8],
-		) -> Weight {
-			// We can use whatever asset here
-			let asset = MultiLocation::parent();
-
-			let call_bytes: Vec<u8> =
-				dest.clone()
-					.encode_call(UtilityAvailableCalls::AsDerivative(
-						index.to_owned(),
-						call.to_owned(),
-					));
-
-			Self::weight_of_transact(
-				&asset,
-				&dest.clone().destination(),
-				weight,
-				call_bytes,
-				OriginKind::SovereignAccount,
-			)
-		}
+		/// Returns weight of `weight_of_initiate_reserve_withdraw` call.
+		fn weight_of_initiate_reserve_withdraw() -> Weight {
+			let dest = MultiLocation::parent();
 
-		/// Returns weight of `transact_through_sovereign call.
-		fn weight_of_transact_through_sovereign(
-			weight: &u64,
-			call: &Vec<u8>,
-			origin_kind: OriginKind,
-		) -> Weight {
+			// We can use whatever asset here
 			let asset = MultiLocation::parent();
-			let dest = MultiLocation::parent();
-			Self::weight_of_transact(&asset, &dest, weight, call.clone(), origin_kind)
-		}
 
-		/// Returns weight of transact message.
-		fn weight_of_transact(
-			asset: &MultiLocation,
-			dest: &MultiLocation,
-			weight: &u64,
-			call: Vec<u8>,
-			origin_kind: OriginKind,
-		) -> Weight {
 			// Construct MultiAsset
 			let fee = MultiAsset {
 				id: Concrete(asset.clone()),
 				fun: Fungible(0),
 			};
 
-			if let Ok(msg) = Self::transact_message(
-				dest.clone(),
-				fee.clone(),
-				weight.clone(),
-				call.clone(),
-				weight.clone(),
-				origin_kind,
-			) {
-				T::Weigher::weight(&mut msg.into()).map_or(Weight::max_value(), |w| {
-					T::BaseXcmWeight::get().saturating_add(w)
-				})
-			} else {
-				0
-			}
+			let xcm: Xcm<()> = Xcm(vec![
+				WithdrawAsset(fee.into()),
+				InitiateReserveWithdraw {
+					assets: MultiAssetFilter::Wild(All),
+					reserve: dest.clone(),
+					xcm: Xcm(vec![]),
+				},
+			]);
+			T::Weigher::weight(&mut xcm.into()).map_or(Weight::max_value(), |w| {
```

Comment:
I hadn't really looked into `Weigher` or `FixedWeightBounds` before... the latter strikes me as rather heavy, to the extent that I almost want to benchmark our `weight` fns here :sweat_smile: